File: service-mvc/src/main/resources/archetype-resources/src/test/java/MainTest.java
## Licensed under the Apache License, Version 2.0 (the "License");
## you may not use this file except in compliance with the License.
## You may obtain a copy of the License at
##
## http://www.apache.org/licenses/LICENSE-2.0
##
## Unless required by applicable law or agreed to in writing, software
## distributed under the License is distributed on an "AS IS" BASIS,
## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
## See the License for the specific language governing permissions and
## limitations under the License.
package ${package};

import static org.assertj.core.api.Assertions.assertThat;

import javax.inject.Inject;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.test.context.junit4.SpringRunner;

import  $

import com.opentable.Main;{package}.Main;

@RunWith(SpringRunner.class)
@SpringBootTest(classes= Main.class, webEnvironment=WebEnvironment.RANDOM_PORT)
public class MainTest {

    @Inject
    private TestRestTemplate testRestTemplate;

    @Test
    public void testHelloWorld() {
        ResponseEntity<String> actualResponse = testRestTemplate.getForEntity("/", String.class);
        assertThat(actualResponse.getStatusCode()).isEqualTo(HttpStatus.OK);
        assertThat(actualResponse.getBody()).isEqualTo("Hello, world!");
    }

}

File: service-mvc-kotlin/src/main/resources/archetype-resources/src/test/kotlin/MainTest.kt
## Licensed under the Apache License, Version 2.0 (the "License");
## you may not use this file except in compliance with the License.
## You may obtain a copy of the License at
##
## http://www.apache.org/licenses/LICENSE-2.0
##
## Unless required by applicable law or agreed to in writing, software
## distributed under the License is distributed on an "AS IS" BASIS,
## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
## See the License for the specific language governing permissions and
## limitations under the License.
package ${package};

import org.assertj.core.api.Assertions.assertThat

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith

import org.springframework.beans.factory.annotation.Autowired

import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment

import org.springframework.boot.test.web.client.TestRestTemplate

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity

import org.springframework.test.context.junit.jupiter.SpringExtension

@ExtendWith(SpringExtension::class)
@SpringBootTest(classes = [Main::class], webEnvironment = WebEnvironment.RANDOM_PORT)
class MainTest(@Autowired val testRestTemplate: TestRestTemplate) {
    @Test
    fun testHelloWorld() {
        val actualResponse: ResponseEntity<String> = testRestTemplate.getForEntity("/", String::class.java)
        assertThat(actualResponse.statusCode).isEqualTo(HttpStatus.OK)
        assertThat(actualResponse.body).isEqualTo("Hello, world!")
    }
}
